### DISCUSSION TITLE
Call rest api to create view and serve it with a slugifyed url

### DISCUSSION THREAD

**Discussion by blueray453** [fastapi/fastapi#5188] [Questions]
Can someone please give me an working example of :

when i hit `examples.com/my-12th-post` will create view (HTML+CSS) in the server using `api.examples.com/post/12`  (where `api.examples.com/post/12` is a rest api).

Here, `my-12th-post` is the slug of `post/12`'s headline.

Here, I want two files. One will create the Rest API. The other is for SSR (server side rendering) and slugifying the URL.

rest-api.py_
```
@app.get("/post/{item_id}")
def func1():
    // Output in JSON when request api.examples.com/post/12
```
_server-side-renderer.py_
```
@app.get(" /my-12th-post")
def func_rend_func1():
    // Output HTML using func1 when request /my-12th-post
```

**Comment by JarroVGIT** [comment]
You would make the first one with FastAPI but I wouldn't recommend to create the SSR with FastAPI. 

The package is built as a high performant API framework. For rendering frontends, you could create another API endpoint (/slug/{id}) which returns the slug, which your frontend framework (React or Vue or something) can set as current URL.
